File: backend/src/Store.Domain/Validators/ProductValidator.cs
using FluentValidation;
using Store.Domain.Entities;

namespace Store.Domain.Validators;

public class ProductValidator : AbstractValidator<Product>
{
    public ProductValidator()
    {
        RuleFor(product => product.Price).GreaterThan(0).WithMessage("O preço deve ser maior que zero.");

        RuleFor(product => product.Title).NotEmpty().WithMessage("O título é obrigatório.");

        RuleFor(product => product.Description).NotEmpty().WithMessage("A descrição é obrigatória.");
    }
}

File: frontend/src/shared/domain-types/validators/product-validator.ts
import * as yup from 'yup';

export const productFormValidation = yup.object({
  price: yup.number()
    .typeError('O preço deve ser um número válido')
    .moreThan(0, 'O preço deve ser maior que zero')
    .required('O campo é obrigatório'),
  title: yup.string().required('O campo é obrigatório'),
  description: yup.string().required('O campo é obrigatório'),
});